pkg/cmd/cli/cmd/newapp.go

```diff
@@ -511,6 +521,20 @@ func setLabels(labels map[string]string, result *newcmd.AppResult, ignoreFailure
 	return nil
 }
 
+func hasLabel(labels map[string]string, result *newcmd.AppResult) (bool, error) {
+	for _, obj := range result.List.Items {
+		objCopy, err := kapi.Scheme.DeepCopy(obj)
+		if err != nil {
+			return false, err
+		}
+		err = util.AddObjectLabelsWithFlags(objCopy.(runtime.Object), labels, util.ErrorOnExistingDstKey)
```

Comment:
Of course if you do change that, then the method name should also change to hasDifferentLabels (or something like that)